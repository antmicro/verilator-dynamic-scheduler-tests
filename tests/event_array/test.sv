module t;
    localparam N = 100;

    event events[N-1:0];

    initial begin
        for (int i = 0; i < N; i++) begin
            @events[i];
            $write("Triggered %d\n", i);
        end
        $write("*-* All Finished *-*\n");
        $finish;
    end

    initial begin
        for (int i = 0; i < N; i++)
            #1 ->events[i];
    end
endmodule
