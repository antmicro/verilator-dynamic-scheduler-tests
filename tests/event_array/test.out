Triggered           0
Triggered           1
Triggered           2
Triggered           3
Triggered           4
Triggered           5
Triggered           6
Triggered           7
Triggered           8
Triggered           9
Triggered          10
Triggered          11
Triggered          12
Triggered          13
Triggered          14
Triggered          15
Triggered          16
Triggered          17
Triggered          18
Triggered          19
Triggered          20
Triggered          21
Triggered          22
Triggered          23
Triggered          24
Triggered          25
Triggered          26
Triggered          27
Triggered          28
Triggered          29
Triggered          30
Triggered          31
Triggered          32
Triggered          33
Triggered          34
Triggered          35
Triggered          36
Triggered          37
Triggered          38
Triggered          39
Triggered          40
Triggered          41
Triggered          42
Triggered          43
Triggered          44
Triggered          45
Triggered          46
Triggered          47
Triggered          48
Triggered          49
Triggered          50
Triggered          51
Triggered          52
Triggered          53
Triggered          54
Triggered          55
Triggered          56
Triggered          57
Triggered          58
Triggered          59
Triggered          60
Triggered          61
Triggered          62
Triggered          63
Triggered          64
Triggered          65
Triggered          66
Triggered          67
Triggered          68
Triggered          69
Triggered          70
Triggered          71
Triggered          72
Triggered          73
Triggered          74
Triggered          75
Triggered          76
Triggered          77
Triggered          78
Triggered          79
Triggered          80
Triggered          81
Triggered          82
Triggered          83
Triggered          84
Triggered          85
Triggered          86
Triggered          87
Triggered          88
Triggered          89
Triggered          90
Triggered          91
Triggered          92
Triggered          93
Triggered          94
Triggered          95
Triggered          96
Triggered          97
Triggered          98
Triggered          99
*-* All Finished *-*
